src/scripts.js

```diff
@@ -1,41 +1,77 @@
 
+const favButton = document.querySelector('.view-favorites')
 let cardArea = document.querySelector('.all-cards');
 let favoriteButton = document.querySelector('.favorite');
 let addButton = document.querySelector('.add')
+let cookbook = new Cookbook(recipeData);
 let user;
 
 window.onload = onStartup();
 
+favButton.addEventListener('click', viewFavorites)
+cardArea.addEventListener("click", buttonConditionals);
+
 function onStartup() {
   let userId = (Math.floor(Math.random() * 49) + 1)
   let newUser = users.find(user => {
     return user.id === Number(userId);
   });
   user = new User(userId, newUser.name, newUser.pantry)
-  populateCards(recipeData);
+  populateCards(cookbook.recipes);
+  greetUser();
+}
+
+function viewFavorites() {
+  cardArea.innerHTML = '';
+  user.favoriteRecipes.forEach(recipe => {
+    cardArea.insertAdjacentHTML("afterbegin", `<div class='card'>
+        <header class='card-header'>
+          <img class='add card-button' src='https://image.flaticon.com/icons/svg/32/32339.svg' alt='Add to recipes to cook'>
+          <button id='${recipe.id}' class='favorite favorite-active card-button' alt='Favorite this recipe'></button>
+        </header>
+          <span class='recipe-name'>${recipe.name}</span>
+          <img class='card-picture' src="${recipe.image}" alt="Food from recipe">
+    </div>`)
+  });
+};
+
+function greetUser() {
+  const userName = document.querySelector('.user-name');
+  userName.innerHTML = user.name;
+
 }
 
-cardArea.addEventListener("click", buttonConditionals);
 
 function favoriteCard(event) {
-  event.target.classList.add('favorite-active');
-  user.favoriteRecipes.push('hello')
-  console.log(user.favoriteRecipes)
+  let specificRecipe = cookbook.recipes.find(recipe => {
+    // console.log(event.target.id);
+    if (recipe.id === Number(event.target.id)) {
+      return recipe;
+    }
+  })
+  if (!event.target.classList.contains('favorite-active')) {
+    event.target.classList.add('favorite-active');
+    console.log(specificRecipe);
+    user.addToFavorites(specificRecipe);
+  } else if (event.target.classList.contains('favorite-active')) {
+    event.target.classList.remove('favorite-active');
+    user.removeFromFavorites(specificRecipe)
+  }
 }
 
 function buttonConditionals(event) {
```

Comment:
Could you explain where event is coming from?  I may have forgotten about this since last inning, but on line 12: `cardArea.addEventListener("click", buttonConditionals);` it seems to me that the function `buttonConditionals` is not being invoked with parentheses, and is lacking the argument for `event` in it's declaration.  But again, I may be forgetting something here, so please correct me if that's the case.

Reply:
The event in the parens of the function exists strictly for developer empathy. It lets other people know that there is an event happening. It would still work if the parens were empty but that is bad convention. I don't remember why you don't need the parens in the event listener but I can see in my past projects that this is the correct syntax.